admin/app/jobs/MissingVideoEncodingsJob.scala

```diff
@@ -0,0 +1,105 @@
+package jobs
+
+import common.{Edition, Logging, ExecutionContexts, AkkaAgent}
+import scala.language.postfixOps
+import conf.{Switches, LiveContentApi}
+import LiveContentApi.getResponse
+import com.gu.contentapi.client.GuardianContentApiError
+import com.gu.contentapi.client.model.Content
+import scala.xml.XML
+import play.api._
+
+import scala.util.{Failure, Success}
+import scala.concurrent.{Future, Await}
+import scala.concurrent.duration._
+import akka.util.Timeout
+import org.joda.time.Seconds
+import play.api.libs.ws.WS
+import play.api.Play.current
+import services.MissingVideoEncodings
+import model.diagnostics.video.DynamoDbStore
+
+
+object VideoEncodingsJob extends ExecutionContexts with Logging  {
+
+  private val videoEncodingsAgent = AkkaAgent[Map[String, List[(String, String, String)]]](Map.empty)
+  implicit val timeout = Timeout(5 seconds)
+
+  def getReport(report: String): List[(String, String, String)] = videoEncodingsAgent().get(report).getOrElse(List(("Not","Yet","Ready")))
```

Comment:
Aha ... I see. I think it would be better to have an `Option[List[(String, String, String)]]`

And maybe it would be better not to have a tuple - you could make your own case class here. I don't know what any of those strings are supposed to represent reading this code.